src/layer.js

```diff
@@ -327,46 +347,22 @@ export class MapLayer extends HTMLElement {
   zoomTo() {
     if(!this.extent) return;
     let map = this._layer._map,
-      tL = this.extent.topLeft.pcrs,
-      bR = this.extent.bottomRight.pcrs,
-      layerBounds = L.bounds(L.point(tL.horizontal, tL.vertical), L.point(bR.horizontal, bR.vertical)),
-      center = map.options.crs.unproject(layerBounds.getCenter(true)),
-      newZoom = map.getZoom();
-
-    let maxZoom = this.extent.zoom.maxZoom, minZoom = this.extent.zoom.minZoom;
-
-    let scale = map.options.crs.scale(newZoom),
-      mapCenterTCRS = map.options.crs.transformation.transform(layerBounds.getCenter(true), scale);
-
-    let mapHalf = map.getSize().divideBy(2),
-      mapTlNew = mapCenterTCRS.subtract(mapHalf).round(),
-      mapBrNew = mapCenterTCRS.add(mapHalf).round();
-
-    let mapTlPCRSNew = M.pixelToPCRSPoint(mapTlNew, newZoom, map.options.projection),
-      mapBrPCRSNew = M.pixelToPCRSPoint(mapBrNew, newZoom, map.options.projection);
-
-    let mapPCRS = L.bounds(mapTlPCRSNew, mapBrPCRSNew);
-
-    let zOffset = mapPCRS.contains(layerBounds) ? 1 : -1;
+        tL = this.extent.topLeft.pcrs,
+        bR = this.extent.bottomRight.pcrs,
+        layerBounds = L.bounds(L.point(tL.horizontal, tL.vertical), L.point(bR.horizontal, bR.vertical)),
+        center = map.options.crs.unproject(layerBounds.getCenter(true));
 
-    while((zOffset === -1 && !(mapPCRS.contains(layerBounds)) && (newZoom - 1) >= minZoom)  ||
-          (zOffset === 1 && mapPCRS.contains(layerBounds) && (newZoom + 1) <= maxZoom)) {
-      newZoom += zOffset;
-
-      scale = map.options.crs.scale(newZoom);
-      mapCenterTCRS = map.options.crs.transformation.transform(layerBounds.getCenter(true), scale);
-
-      mapTlNew = mapCenterTCRS.subtract(mapHalf).round();
-      mapBrNew = mapCenterTCRS.add(mapHalf).round();
-      mapTlPCRSNew = M.pixelToPCRSPoint(mapTlNew, newZoom, map.options.projection);
-      mapBrPCRSNew = M.pixelToPCRSPoint(mapBrNew, newZoom, map.options.projection);
-
-      mapPCRS = L.bounds(mapTlPCRSNew, mapBrPCRSNew);
-    }
-    if(zOffset === 1 && newZoom - 1 >= 0) newZoom--;
-    map.setView(center, newZoom, {animate: false});
+    let maxZoom = this.extent.zoom.maxZoom, 
+        minZoom = this.extent.zoom.minZoom;
+    map.setView(center, M.getMaxZoom(layerBounds, map, minZoom, maxZoom), {animate: false});
   }
   mapml2geojson(options = {}){
     return M.mapml2geojson(this, options);
   }
+  _pasteFeature(text) {
```

Comment:
when pasting features to a layer using this method or pasting through the console, using layer.zoomTo() still zooms to the old bbox, maybe that stuff should be updated when new features are pasted in.

i.e.
```js
let feature = "<map-feature><map-featurecaption>Ottawa</map-featurecaption><map-geometry><map-point class='ottawa'><map-coordinates>-75.697193 45.421530</map-coordinates></map-point></map-geometry><map-properties><table><thead><tr><th role='columnheader' scope='col'>Property name</th><th role='columnheader' scope='col'>Property value</th></tr></thead><tbody><tr><th scope='row'>Name</th><td itemprop='amenity'>Ottawa</td></tr><tr><th scope='row'>Type</th><td itemprop='name'>City</td></tr><tr><th scope='row'>Website</th><td itemprop='website'><a href='https://ottawa.ca/' target='_blank'>Ottawa</a></td></tr></tbody></table></map-properties></map-feature>"

layer._pasteFeature(feature);
```

P.S. we should rename this so it's not a 'private' method.

Reply:
I think this should also allow pasting of not only text, but also map-feature elements.

Reply:
When editing the coordinates of a feature in light DOM, the `<map-feature>` element disappears, but the marker does move.